PantheonCMD/pcbuild.py

```diff
@@ -82,28 +98,52 @@ def prepare_build_directory():
     os.makedirs('build/images')
 
 
-def coalesce_document(main_file, attributes=None):
+def resolve_attribute_tree(content, attributes):
+    """Attempt to recursively resolve AsciiDoc attributes into a string."""
+
+    continue_processing = True
+
+    while continue_processing:
+        if re.match(r'.*\{\S+\}.*', content):
+            for attribute in re.search(r'\{(.*?)\}', content).groups():
+                if attribute in attributes.keys():
+                    content = content.replace('{' + attribute + '}',attributes[attribute])
+                else:
+                    continue_processing = False
+        else:
+            continue_processing = False
+    return content
+
+
+def coalesce_document(main_file, attributes=None, depth=1, top_level=True):
     """Combines the content from includes into a single, contiguous source file."""
     attributes = attributes or {}
     lines = []
 
     # Open the file, iterate over lines
     if os.path.exists(main_file):
         with open(main_file) as input_file:
+            # Iterate over content
             for line in input_file:
                 # Process comments
                 if line.startswith('//'):
```

Comment:
Will this catch lines in multi line comments?

Reply:
Thanks @Levi-Leah - added some logic for this so that, if the script encounters a comment block boundary, it checks if it's a start or finish, and ignores anything in between.